integration_tests/calibration.cpp

```diff
@@ -7,9 +7,8 @@
 using namespace dronecode_sdk;
 using namespace std::placeholders; // for `_1`
 
-static void receive_calibration_callback(Calibration::Result result,
-                                         float progress,
-                                         const std::string text,
+static void receive_calibration_callback(const Calibration::Result result,
+                                         const Calibration::ProgressData &progress_data,
```

Comment:
Why are some references and some not?

Reply:
Good point, I guess they should all be references... unless you have a different opinion?

Reply:
I would say normal by-value for all.

Reply:
Disagreed, [see here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const): "For “in” parameters, pass cheaply-copied types by value and others by reference to const".

Reply:
Right, unless you use `std::move`.

Reply:
I'm not sure what you mean. So you make the function take them by value, and assume that the caller moves them? Or you take them by value, but move them in the implementation of the function?

In any case I am not sure it behaves exactly the same as a const reference, does it? And a const ref is actually extremely clear on the intention ("I will just read that thing").

Reply:
Ok, you're right. So I vote `Calibration::Result result` and `Calibration::ProgressData &progress_data`.

Reply:
Deal. Done :laughing:

Reply:
Oh I now realize this is just the callback used in the integration test :man_facepalming:.